src/radical/pilot/compute_pilot.py

```diff
@@ -156,10 +153,12 @@ def _update(self, pilot_dict):
 
 
             try:
-                assert(rps._pilot_state_value(target) - rps._pilot_state_value(current)), \
-                            'invalid state transition'
-            except:
-                self._log.error('%s: invalid state transition %s -> %s', 
+                state_diff = rps._pilot_state_value(target) - \
+                             rps._pilot_state_value(current)
+                assert(state_diff), 'invalid state transition'
```

Comment:
Hello @andre-merzky , I tried:
```python
>>> assert(2)
>>> try:assert(2)
... except: print 'False'
...
>>> try:assert(-1)
... except: print 'False'
...
>>> try:assert(0)
... except: print 'False'
...
False
```

Will the above behave as expected?

Reply:
Shouldn't it be:
```python
assert(state_diff == 1)
```

Reply:
Ah, indeed - the test should be `assert(state_diff >= 1`, will fix!

Reply:
`== 1` was correct - no gaps are allowed at that point.